packages/astro/src/core/render/route-cache.ts

```diff
@@ -58,6 +58,21 @@ export async function callGetStaticPaths({
 		// Q: Why the cast?
 		// A: So users downstream can have nicer typings, we have to make some sacrifice in our internal typings, which necessitate a cast here
 		paginate: generatePaginateFunction(route) as PaginateFunction,
+		pathToParam(path: string) {
```

Comment:
Is there a case where one would not want to use `pathToParam`, and pass the string directly? And if they do so, wouldn't they hit issues as described in the linked issue? If we see that the params should always be normalized, maybe we should do that for all params automatically.

Reply:
The assumption is that every param should be a valid URL, that's the same assumption we're using as pillar before discussing any bug that affects these issues e.g. encoding. 

Regarding the _why_, it's exactly why Matthew decided to create this utility in the first place. We can't predict the usage of users, and felt that it's better to provide an API instead of forcing our way around (always regarding the first assumption)

Reply:
If we don't force our way around, or to put it differently, if the original issue author decided that they don't want to use `pathToParam` because that's what they want, wouldn't that not fix the issue?

The root issue in https://github.com/withastro/astro/issues/9946 was that the sitemap integration wasn't constructing the paths right because of the `//`. I'm guessing it's this issue:

```js
new URL('//blog/', 'http://example.com') // -> http://blog/
```

So if we want to allow users to use slashes still and fix the root issue, the sitemap integration would need a fix. And in that case, I'm not sure if there's any other use for `pathToParam`.

Reply:
The more I think about it the more I question this helper as being useful for SSG. The example use-case of `/about//en` is a valid URL, so I wouldn't want to block anyone from using it (most likely as a legacy URL). But in the context of SSG, is that even possible to have. I think we need to do some more investigation.